lib/std/date_time.zig

```diff
@@ -0,0 +1,160 @@
+pub fn DateTime(comptime DateT: type, comptime TimeT: type) type {
+    return struct {
+        date: Date,
+        time: Time = .{},
+
+        pub const Date = DateT;
+        pub const Time = TimeT;
+        /// Fractional epoch seconds based on `TimeT.precision`:
+        ///   0 = seconds
+        ///   3 = milliseconds
+        ///   6 = microseconds
+        ///   9 = nanoseconds
+        pub const EpochSubseconds = std.meta.Int(
+            @typeInfo(Date.EpochDays).Int.signedness,
+            @typeInfo(Date.EpochDays).Int.bits + std.math.log2_int_ceil(usize, Time.fs_per_day),
+        );
+
+        const Self = @This();
+
+        /// New date time from fractional epoch seconds.
+        pub fn fromEpoch(subseconds: EpochSubseconds, time_opts: Time.Options) Self {
+            const days = @divFloor(subseconds, s_per_day * Time.fs_per_s);
+            const new_date = Date.fromEpoch(@intCast(days));
+            const day_seconds = std.math.comptimeMod(subseconds, s_per_day * Time.fs_per_s);
+            const new_time = Time.fromDaySeconds(day_seconds, time_opts);
+            return .{ .date = new_date, .time = new_time };
+        }
+
+        /// Returns fractional epoch seconds.
+        pub fn toEpoch(self: Self) EpochSubseconds {
```

Comment:
IMHO, this could be more specific - Unix is not the only one... (or allow a user-defined epoch)

Reply:
`epoch` is a user-defined generic parameter to `Date`. I'll make the comment explicitly mention this.

Reply:
It's pretty common for different systems. See the existing `time/epoch.zig`.